server/api/tumdev/campus_backend.proto

```diff
@@ -564,7 +564,17 @@ message GetCanteenHeadCountReply {
   // A time stamp indicating how up to date the response is. Only valid in case percent != -1.
   google.protobuf.Timestamp timestamp = 4;
 }
-message ListStudentClubRequest {}
+
+enum Language {
+  German = 0;
+  English = 1;
+}
+
+message ListStudentClubRequest {
+  // Language of the student clubs and categories
+  // Defaults to german
+  optional Language language = 1;
```

Comment:
Out of interest: isn't the default language set to English here? Or has that no effect? 😄

Reply:
No, `German = 0` is the default for the `enum Language`.

What you are confused by is the position-designator (maybe not the correct term) in the protobuf (the `= 1`).

See https://protobuf.dev/programming-guides/proto3/#default

> When a message is parsed, if the encoded message does not contain a particular implicit presence element, accessing the corresponding field in the parsed object returns the default value for that field. These defaults are type-specific:
> 
> - For strings, the default value is the empty string.
> - For bytes, the default value is empty bytes.
> - For bools, the default value is false.
> - For numeric types, the default value is zero.
> - For enums, the default value is the first defined enum value, which must be 0.
> - For message fields, the field is not set. Its exact value is language-dependent.
> - See the [generated code guide](https://protobuf.dev/reference/) for details.
The default value for repeated fields is empty (generally an empty list in the appropriate language).
>
> Note that for scalar message fields, once a message is parsed there’s no way of telling whether a field was explicitly set to the default value (for example whether a boolean was set to false) or just not set at all: you should bear this in mind when defining your message types. For example, don’t have a boolean that switches on some behavior when set to false if you don’t want that behavior to also happen by default. Also note that if a scalar message field is set to its default, the value will not be serialized on the wire. If a float or double value is set to +0 it will not be serialized, but -0 is considered distinct and will be serialized.
> 
> See the [generated code guide](https://protobuf.dev/reference/) for your chosen language for more details about how defaults work in generated code.

Reply:
and yes: I also came across this behaviour mostly by acident, by looking into what the `GetLanguage` function does ^^

=> this was a good question